java/client/src/main/java/glide/api/commands/GeospatialIndicesBaseCommands.java

```diff
@@ -135,4 +139,270 @@ CompletableFuture<Long> geoadd(
+     *
+     * @param resultOptions Contains up to 2 optional input:
+     *     <ul>
+     *       <li>SortOrder: The query's order to sort the results by:
+     *           <ul>
+     *             <li>ASC: Sort returned items from the nearest to the farthest, relative to the
+     *                 center point.
+     *             <li>DESC: Sort returned items from the farthest to the nearest, relative to the
+     *                 center point.
+     *           </ul>
+     *       <li>COUNT: Limits the results to the first N matching items, when the 'ANY' option is
+     *           used, the command returns as soon as enough matches are found. This means that the
+     *           results might may not be the ones closest to the origin.
+     *     </ul>
+     *
+     * @return An array of arrays where each sub-array represents a single item in the following
+     *     order:
+     *     <ul>
+     *       <li>The member (location) name.
+     *       <li>The distance from the center as a floating point number, in the same unit specified
+     *           for <code>searchBy</code>.
+     *       <li>The member (location) name.
+     *       <li>The distance from the center as a floating point number, in the same unit specified
+     *           for <code>searchBy</code>.
+     *       <li>The Geohash of the location.
+     *       <li>The coordinates as a two item array
+     *     </ul>
+     *
+     * @example
+     *     <pre>{@code
+     * Object[] expected =
+     *            new Object[] {
+     *                new Object[] {
+     *                    "Palermo",
+     *                    new Object[] {
+     *                        0.0, 3479099956230698L, new Object[] {13.361389338970184, 38.1155563954963}
+     *                    }
+     *                },
+     *                new Object[] {
+     *                    "Catania",
+     *                    new Object[] {
+     *                        166.2742, 3479447370796909L, new Object[] {15.087267458438873, 37.50266842333162}
+     *                    }
+     *                }
+     *            };
+     * Object[] result =
+     *                client
+     *                        .geosearch(
+     *                                "a1",
+     *                                new GeoSearchOrigin("Palermo"),
+     *                                new GeoSearchShape(200, GeoUnit.KILOMETERS),
+     *                                new GeoSearchOptions.GeoSearchOptionsBuilder()
+     *                                             .withcoord()
+     *                                             .withdist()
+     *                                             .withhash()
+     *                                             .count(3)
+     *                                             .build(),
+     *                                SortOrder.ASC)
+     *                        .get();
+     * System.out.printf("Geosearch result:" + Arrays.toString(objects));
```

Comment:
Maybe you can replace this line with a comment which says
> The `result` contains the data in the same format as `expected`.

You can also add inline code comments to fields in expected, like
```
1.2, // the distance
100542L // the geohash
```